mycpp/leaky_types.cc

```diff
@@ -2,16 +2,144 @@
   #include "mycpp/mylib_old.h"
 using gc_heap::StackRoots;  // no-op
 using mylib::AllocStr;
+using mylib::CopyBufferIntoNewStr;
 #else
   #include "mycpp/gc_builtins.h"
   #include "mycpp/gc_types.h"
 using gc_heap::kEmptyString;
 using gc_heap::StackRoots;
 using gc_heap::Str;
+using gc_heap::CopyBufferIntoNewStr;
 #endif
 
 #include <ctype.h>  // isalpha(), isdigit()
 
+// Translation of Python's print().
+void print(Str* s) {
+  fputs(s->data(), stdout);
+  fputs("\n", stdout);
+}
+
+// Like print(..., file=sys.stderr), but Python code explicitly calls it.
+void println_stderr(Str* s) {
+  fputs(s->data(), stderr);
+  fputs("\n", stderr);
+}
+
+Str* str_repeat(Str* s, int times) {
+  // Python allows -1 too, and Oil used that
+  if (times <= 0) {
+    return kEmptyString;
+  }
+  int len_ = len(s);
+  int new_len = len_ * times;
+  char* data = static_cast<char*>(malloc(new_len + 1));
```

Comment:
I think this should now use `AllocStr(new_len)` and then copy into that buffer, but since we're doing mechanical transformations now, this is fine

Although I don't see what the point of CopyBufferIntoNewStr() is, because StrFromC() should suffice?  A buffer that is malloc'd is "from C", i.e. it is not garbage collected yet